src/compressor.rs

```diff
@@ -141,6 +141,23 @@ impl<'a> Compressor<'a> {
         pb.enable_steady_tick(100);
 
         for (&state_group, entry) in self.original_state_map {
+            // Check whether this entry is in_range or is just present in the map due to being
+            // a predecessor of a group that IS in_range for compression
+            if !entry.in_range {
```

Comment:
We don't ever seem to set this to false anywhere?

Reply:
we create StateGroupEntries with default() which sets it to false, and then it gets explicity sets to true for some state groups

Reply:
It's only the state groups loaded with get_missing_from_db that can not be in range (i.e. groups just loaded because they're the predecessors of groups that are in range)

Reply:
Aha! Makes sense, thanks